test/integration/im_cloudbuild_workspace_gitlab/im_cloudbuild_workspace_gitlab_test.go

```diff
@@ -0,0 +1,331 @@
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/stretchr/testify/assert"
+	"github.com/xanzy/go-gitlab"
+)
+
+type GitLabClient struct {
+	t         *testing.T
+	client    *gitlab.Client
+	group     string
+	namespace int
+	repo      string
+	project   *gitlab.Project
+}
+
+func NewGitLabClient(t *testing.T, token, owner, repo string) *GitLabClient {
+	t.Helper()
+	client, err := gitlab.NewClient(token)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	return &GitLabClient{
+		t:         t,
+		client:    client,
+		group:     "infrastructure-manager",
+		namespace: 84326276,
+		repo:      repo,
+	}
+}
+
+func (gl *GitLabClient) ProjectName() string {
+	return fmt.Sprintf("%s/%s", gl.group, gl.repo)
+}
+
+func (gl *GitLabClient) GetProject() *gitlab.Project {
+	proj, resp, err := gl.client.Projects.GetProject(gl.ProjectName(), nil)
+	if resp.StatusCode != 404 && err != nil {
+		gl.t.Fatalf("got status code %d, error %s", resp.StatusCode, err.Error())
+	}
+	gl.project = proj
+	return proj
+}
+
+func (gl *GitLabClient) CreateProject() {
+	opts := &gitlab.CreateProjectOptions{
+		Name:        gitlab.Ptr(gl.repo),
+		NamespaceID: gitlab.Ptr(84326276),
+		// Required otherwise Cloud Build errors on creating the connection
+		InitializeWithReadme: gitlab.Ptr(true),
+	}
+	proj, _, err := gl.client.Projects.CreateProject(opts)
+	if err != nil {
+		gl.t.Fatal(err.Error())
+	}
+	gl.project = proj
+}
+
+func (gl *GitLabClient) AddFileToProject(file []byte) {
+	opts := &gitlab.CreateFileOptions{
+		Branch:        gitlab.Ptr("main"),
+		CommitMessage: gitlab.Ptr("Initial config commit"),
+		Content:       gitlab.Ptr(string(file)),
+	}
+	_, _, err := gl.client.RepositoryFiles.CreateFile(gl.ProjectName(), "main.tf", opts)
+	if err != nil {
+		gl.t.Fatal(err.Error())
+	}
+}
+
+func (gl *GitLabClient) DeleteProject() {
+	resp, err := gl.client.Projects.DeleteProject(gl.ProjectName())
+	if err != nil {
+		gl.t.Errorf("error deleting project with status %s and error %s", resp.Status, err.Error())
+	}
+	gl.project = nil
+}
+
+// GetOpenMergeRequest gets an open merge request for a given branch if it exists.
+func (gl *GitLabClient) GetOpenMergeRequest(branch string) *gitlab.MergeRequest {
+	state := "opened"
+	opts := gitlab.ListProjectMergeRequestsOptions{
+		State:        &state,
+		SourceBranch: &branch,
+	}
+	mergeRequests, _, err := gl.client.MergeRequests.ListProjectMergeRequests(gl.ProjectName(), &opts)
+	if err != nil {
+		gl.t.Fatalf(err.Error(), err)
+	}
+	if len(mergeRequests) == 0 {
+		return nil
+	}
+	return mergeRequests[len(mergeRequests)-1]
+}
+
+func (gl *GitLabClient) CreateMergeRequest(title, branch, base string) *gitlab.MergeRequest {
+	opts := gitlab.CreateMergeRequestOptions{
+		Title:        &title,
+		SourceBranch: &branch,
+		TargetBranch: &base,
+	}
+	mergeRequest, _, err := gl.client.MergeRequests.CreateMergeRequest(gl.ProjectName(), &opts)
+	if err != nil {
+		gl.t.Fatalf(err.Error(), err)
```

Comment:
here and throughout, this seems odd we are passing in error.Error + err as arg.
```suggestion
		gl.t.Fatalf(err.Error())
```